src/dash-table/dash/DataTable.js

```diff
@@ -955,6 +958,43 @@ export const propTypes = {
         )
     ),
 
+    /**
+     * `tooltip_header` represents the tooltip shown
+     * for different header columns and cells.
+     * The `property` name refers to the column ID. Each property
+     * contains a list of tooltips mapped to the table's `header`
+     * row index.
```

Comment:
"Each property contains a list of tooltips" implies to me that we're expecting a structure:
```
{'col-id': ['first row', 'second row']}
```
rather than what I think we're actually expecting:
```
[{'col-id': 'first row'}, {'col-id': 'second row'}]
```
But the former actually feels a bit more intuitive and compact to me, even if it doesn't match `tooltip_data` as closely. It simplifies some simple cases: single row headers, or the same tooltip for every row, you would just provide a string or object for the value, rather than an array of strings or objects. What do you think?

Also minor point: I'd just say "The property name" rather than "The `property` name" - as is I was looking for something called "property" below 😅

Reply:
Agreed. The current description is inaccurate and misleading.

For consistency, I don't foresee any major problem supporting both syntax (array of objects and object of arrays) for both props.

Reply:
https://github.com/plotly/dash-table/pull/831/commits/a2011e0acdeb1ca1da51d5b255d6e1ef95d5eb2c & https://github.com/plotly/dash-table/pull/831/commits/e4433595a01388746905fee36f48d0e239ab050a

Reply:
Hmm if either syntax can cover every use case, I don't think it's a good idea to support both, that'll just confuse people. 

Seems to me the `*_data` props should all match `data` itself, ie `arrayOf(objectOf(...))`.

For `tooltip_header`, I mostly wanted to see if we could simplify the cases of single-row headers and multi-row headers with the same tooltip for all rows, and to do that I was thinking (1) `tooltip_header` would be an `objectOf` the row IDs, and (2) you could choose not to use an array inside that. So if all you want is simple text that applies the same to every row (of if there's only one row) you could use:
```js
{'col1-id': 'all rows for col1', 'col2-id': 'all rows for col2'}
```
or eg markdown for all rows:
```js
{'col1-id': {'type': 'markdown', 'value': '#Wow\nwhat great data!'}}
```
But of course you'd still need the array form when you have multiple rows and you want them to behave differently:
```js
{
  'col1-id': ['first row', 'second row'],
  'col2-id': [{'type': 'markdown', 'value': '#Big'}, {'type': 'markdown', 'value': '####small'}]
}
```

Reply:
Sure.. can update to have:

`tooltip`: objectOf(string | tooltip) w/ **new** `use_with` for (both,data,header)
`tooltip_conditional`: **unchanged** arrayOf(conditionalTooltip)
`tooltip_data`: **unchanged** arrayOf(objectOf(string | tooltip))
`tooltip_header`: **new** objectOf(string | tooltip | arrayOf(null|string|tooltip))`